lib/chai/core/assertions.js

```diff
@@ -1036,23 +1072,28 @@ module.exports = function (chai, _) {
    * @api public
    */
 
-  Assertion.addMethod('property', function (name, val, msg) {
+  function assertProperty (name, val, msg) {
     if (msg) flag(this, 'message', msg);
 
-    var isNested = !!flag(this, 'nested')
-      , isDeep = !!flag(this, 'deep')
+    var isNested = flag(this, 'nested')
+      , isOwn = flag(this, 'own');
+
+    if (isNested && isOwn) {
+      throw new Error('The "nested" and "own" flags cannot be combined.');
+    }
+
+    var isDeep = flag(this, 'deep')
       , descriptor = (isDeep ? 'deep ' : '')
+                   + (isOwn ? 'own ' : '')
                    + (isNested ? 'nested ' : '')
                    + 'property '
       , negate = flag(this, 'negate')
       , obj = flag(this, 'object')
       , pathInfo = isNested ? _.getPathInfo(name, obj) : null
-      , hasProperty = isNested
-        ? pathInfo.exists
-        : _.hasProperty(name, obj)
-      , value = isNested
-        ? pathInfo.value
-        : obj[name];
+      , hasProperty = isOwn ? Object.prototype.hasOwnProperty.call(obj, name)
```

Comment:
Hahahahaha, I see no problem in keeping them there, it's just a matter of personal taste, would you like to keep them?
If you do, please don't be afraid to say it, there's no problem if you prefer that way 😄